src/utils/rest-api.ts

```diff
@@ -0,0 +1,296 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { APP_NAME, getAppName } from './config-params';
+import { store } from '../redux/store';
+import ReconnectingWebSocket, { Event } from 'reconnecting-websocket';
+import { AppState } from '../redux/reducer';
+
+export interface ErrorWithStatus extends Error {
+    status?: number;
+}
+
+export type Url = Exclude<Parameters<typeof fetch>[0], Request>; //string | URL;
```

Comment:
If you have to comment // string | URL it means that it would be clearer if you just define it as string | URL straight away